src/plugins/columnSorting/columnStatesManager.js

```diff
@@ -156,50 +135,72 @@ export class ColumnStatesManager {
   /**
    * Get if particular column is sorted.
    *
-   * @param {number} column Physical column index.
+   * @param {number} column Visual column index.
    * @returns {boolean}
    */
   isColumnSorted(column) {
-    return this.getSortedColumns().includes(column);
+    return isObject(this.sortingStates.getValueAtIndex(this.hot.toPhysicalColumn(column)));
   }
 
   /**
-   * Get states for all sorted columns.
+   * Queue of sort states containing sorted columns and their orders (Array of objects containing `column` and `sortOrder` properties).
+   *
+   * **Note**: Please keep in mind that returned objects expose **visual** column index under the `column` key.
    *
-   * @returns {Array}
+   * @returns {Array<object>}
    */
   getSortStates() {
-    return deepClone(this.sortedColumnsStates);
+    if (this.sortingStates === null) {
+      return [];
+    }
+
+    const sortingStatesQueue = this.sortingStates.getEntries();
+
+    return sortingStatesQueue.map(
+      ([physicalColumn, value]) => ({ column: this.hot.toVisualColumn(physicalColumn), ...value }));
   }
 
   /**
    * Get sort state for particular column. Object contains `column` and `sortOrder` properties.
    *
-   * **Note**: Please keep in mind that returned objects expose **physical** column index under the `column` key.
+   * **Note**: Please keep in mind that returned objects expose **visual** column index under the `column` key.
    *
-   * @param {number} column Physical column index.
+   * @param {number} column Visual column index.
    * @returns {object|undefined}
    */
   getColumnSortState(column) {
     if (this.isColumnSorted(column)) {
-      return deepClone(this.sortedColumnsStates[this.getIndexOfColumnInSortQueue(column)]);
+      const physicalColumn = this.hot.toPhysicalColumn(column);
+
+      const sortingStateWithPriority = this.sortingStates.getValueAtIndex(physicalColumn);
```

Comment:
If condition `isColumnSorted` internally does `isObject(this.sortingStates.getValueAtIndex(this.hot.toPhysicalColumn(column)))` then again we call `toPhysicalColumn`, and another `getValueAtIndex`. Those are cheap but unnecessary.

We can move the check to the end
```
if (isObject(sortingStateWithPriority)) {
  return {
    ...
  }
}
```

Reply:
Changed within https://github.com/handsontable/handsontable/pull/7276/commits/6ec835fef3ffef5d061ac43e1aabc5f5b834c2bf with some other improvements.